base/docker-integration.go

```diff
@@ -0,0 +1,313 @@
+
+	//Define bucket specs
+	bucketSettings := gocb.BucketSettings{
+		FlushEnabled:  true,
+		IndexReplicas: false,
+		Name:          DefaultTestBucketname,
+		Password:      "",
+		Quota:         1024,
+		Replicas:      0,
+		Type:          gocb.BucketType(0),
+	}
+
+	bucketSettings1 := gocb.BucketSettings{
+		FlushEnabled:  true,
+		IndexReplicas: false,
+		Name:          DefaultTestShadowBucketname,
+		Password:      "",
+		Quota:         1024,
+		Replicas:      0,
+		Type:          gocb.BucketType(0),
+	}
+
+	bucketSettings2 := gocb.BucketSettings{
+		FlushEnabled:  true,
+		IndexReplicas: false,
+		Name:          DefaultTestIndexBucketname,
+		Password:      "",
+		Quota:         1024,
+		Replicas:      0,
+		Type:          gocb.BucketType(0),
+	}
+
+	//Define user specs
+	userSettings := gocb.UserSettings{
+		Password: DefaultTestPassword,
+		Roles: []gocb.UserRole{
+			{"bucket_full_access", DefaultTestBucketname},
+			{"bucket_admin", DefaultTestBucketname},
+		},
+	}
+
+	userSettings1 := gocb.UserSettings{
+		Password: DefaultTestShadowPassword,
+		Roles: []gocb.UserRole{
+			{"bucket_full_access", DefaultTestShadowBucketname},
+			{"bucket_admin", DefaultTestShadowBucketname},
+		},
+	}
+
+	userSettings2 := gocb.UserSettings{
+		Password: DefaultTestIndexPassword,
+		Roles: []gocb.UserRole{
+			{"bucket_full_access", DefaultTestIndexBucketname},
+			{"bucket_admin", DefaultTestIndexBucketname},
+		},
+	}
+
+	//Connect to cluster with gocb and create users and buckets
+	cluster, err = gocb.Connect("http://localhost:8091/")
+	fatalError(`Unable to connect to cluster with gocb`, err)
+	manager := cluster.Manager(DefaultCouchbaseAdministrator, DefaultCouchbasePassword)
+
+	err = manager.UpdateBucket(&bucketSettings)
+	fatalError(`Unable to create bucket`, err)
+	err = manager.UpdateBucket(&bucketSettings1)
+	fatalError(`Unable to create bucket`, err)
+	err = manager.UpdateBucket(&bucketSettings2)
+	fatalError(`Unable to create bucket`, err)
+
+	err = manager.UpsertUser("local", DefaultTestUsername, &userSettings)
+	fatalError(`Unable to create user`, err)
+	err = manager.UpsertUser("local", DefaultTestShadowUsername, &userSettings1)
+	fatalError(`Unable to create user`, err)
+	err = manager.UpsertUser("local", DefaultTestIndexUsername, &userSettings2)
+	fatalError(`Unable to create user`, err)
+
+	err = cluster.Close()
+	if err != nil {
+		fmt.Println("Failed to close gocb cluster connection but will continue")
+	}
+
+	areReady := func(url string, checkReady func(reader io.ReadCloser) bool) {
+		for {
+			req, err := http.NewRequest("GET", url, nil)
+			fatalError(`Failed to poll status`, err)
+			req.SetBasicAuth(DefaultCouchbaseAdministrator, DefaultCouchbasePassword)
+			response, err = client.Do(req)
+			fatalError(`Failed to poll status`, err)
+			if checkReady(response.Body) {
+				return
+			}
+		}
+	}
+
+	//Poll buckets to ensure they are all healthy and ready to work
+	bucketReadyCheck := func(reader io.ReadCloser) bool {
```

Comment:
Adding some logging here would be useful to signal that the test is starting up the server

```suggestion
	bucketReadyCheck := func(reader io.ReadCloser) bool {
		fmt.Println("Polling for bucket readiness...")
```